actions/actions.py

```diff
@@ -107,18 +196,31 @@ def restore() -> None:
                                     "{}.key".format(client.common_name))
             with open(cert_file, 'r') as file:
                 cert = file.read()
-                print('writing cert to '
-                      '{}:\n{}'.format(client.common_name, cert))
             with open(key_file, 'r') as file:
                 key = file.read()
             client.set_cert(cert, key)
 
         except FileNotFoundError:
-            print("Cert not found for {}".format(client.common_name))
-
-
-def list_backups() -> None:
-    file_list: List[str] = []
+            if client.cert_type == 'client':
+                cert, key = create_client_certificate(client.common_name)
+                client.set_cert(cert, key)
+            elif client.cert_type == 'server':
+                cert, key = create_server_certificate(client.common_name,
+                                                      client.sans,
+                                                      client.common_name)
+                client.set_cert(cert, key)
+            else:
+                # This use case should not really happen as easyrsa charm
+                # does not support Application type certificates
+                function_fail('Unrecognized certificate request type '
+                              '"{}".'.format(client.cert_type))
+
+    hookenv._run_atexit()
```

Comment:
Is this the correct place to run this? `hookenv._run_atstart()` is called outside of this scope. Should `_run_at_exit` be called in at the end of the main function?

Reply:
This is a good question. I got the placement and usage of both of these function from examples shared by Kubernetes team. In those examples the `_run_atstart()` was run in the global scope, so I'm gonna assume that's correct. Regarding the `_run_atexit()`, the documentation mentions that it should not be run in case that action fails. That's why I didn't put it at the end of the `main`. However I could maybe place into the `else` branch of `try/except` structure that encapsulates action execution.